Add iati-activity/result/reference and test cases

dalepotter opened the pull request:
Fixes #362

dalepotter committed:
Add iati-activity/result/reference and test cases

dalepotter committed:
Add test case for non-permitted child element in result/reference

hayfield commented on tests/should-fail/iati-activities/iati-activity/result/reference/err-not-iati-xml-missing-attribute_reference-element-present-no-required-attributes.xml:
```diff
@@ -0,0 +1,33 @@
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <reference/><!-- FAIL: Optional `reference` element present, but both the mandatory `code` and `vocabulary` attributes are not present. -->
```
If they are tested individually, this test file is not required.

(and if they aren't tested individually, this test file doesn't check the things that each is required)

...and there's no manner in XSD 1.0 to say that one of a pair of attributes is required https://stackoverflow.com/a/763108

dalepotter replied:
Fixed in 1729f95.

hayfield commented on tests/should-pass/iati-activities/iati-activity/result/reference/reference-element-present-with-namespace-element.xml:
```diff
@@ -0,0 +1,35 @@
+        <narrative>Sample text</narrative>
+      </title>
+      <reference vocabulary="xx" code="xx"><!-- PASS: Optional `reference` element present, with two mandatory attributes. -->
+        <example-namespace:namespace-element></example-namespace:namespace-element><!-- Optional namespace element included. -->
```
Could be clearer that this is what is being tested.

dalepotter replied:
I've made changes in 59a684c - is it clearer now?

hayfield reviewed (dismissed):
One redundant test file and a couple of comments that could be a tad clearer.

allthatilk commented on tests/should-pass/iati-activities/iati-activity/result/reference/no-reference-element-present.xml:
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
```
Is this needed? Isn't it covered by the minimal IATI file case?

allthatilk replied:
Nvm, leave for integration branch as needs to be moved to `result` level folder.

dalepotter replied:
Fixed in e23a674 anyway!

dalepotter committed:
Remove redundant test case file

Each required attribute is tested indivually, so there is no need to test that both are not present. Additionally there is no manner in XSD 1.0 to say that one of a pair of attributes is required.

dalepotter committed:
Improve comment for namespace testing

dalepotter committed:
Move test of results with no optional child elements and modify comments accordingly.
